coalib/bears/meta.py

```diff
@@ -14,16 +15,19 @@ class bearclass(type):
     # by default a bear class has no aspects
     aspects = defaultdict(lambda: aspectlist([]))
 
-    def __new__(mcs, clsname, bases, clsattrs, *varargs, aspects=None):
+    def __new__(mcs, clsname, bases, clsattrs, *varargs, aspects=None,
+                languages=None):
         return type.__new__(mcs, clsname, bases, clsattrs, *varargs)
 
-    def __init__(cls, clsname, bases, clsattrs, *varargs, aspects=None):
+    def __init__(cls, clsname, bases, clsattrs, *varargs, aspects=None,
+                 languages=None):
         """
-        Initializes the ``.aspects`` dict on new bear classes from the mapping
-        given to the keyword-only `aspects` argument.
+        Initializes the ``.aspects`` dict and ``.languages`` array on new
+        bear classes from the mapping given to the keyword-only `aspects`
+        and `languages` arguments.
         """
         type.__init__(cls, clsname, bases, clsattrs, *varargs)
         if aspects is not None:
             cls.aspects = defaultdict(
-                lambda: aspectlist([]),
-                ((k, aspectlist(v)) for (k, v) in dict(aspects).items()))
+        if languages is not None:
```

Comment:
@sims1253 https://www.python.org/dev/peps/pep-0008/#programming-recommendations 

Depedending on its type, `languages.__bool__()` could return `False` but `languages` could be nevertheless a sequence of more than 0 elements